src/bijectors/named_bijector.jl

```diff
@@ -0,0 +1,236 @@
+
+# Examples
+```julia-repl
+julia> using Bijectors: NamedBijector, Scale, Exp
+
+julia> b = NamedBijector((a = Scale(2.0), b = Exp()));
+
+julia> x = (a = 1., b = 0., c = 42.);
+
+julia> b(x)
+(a = 2.0, b = 1.0, c = 42.0)
+
+julia> (a = 2 * x.a, b = exp(x.b), c = x.c)
+(a = 2.0, b = 1.0, c = 42.0)
+```
+"""
+struct NamedBijector{names, Bs<:NamedTuple{names}} <: AbstractNamedBijector
+    bs::Bs
+end
+
+names_to_bijectors(b::NamedBijector) = b.bs
+
+@generated function (b::NamedBijector{names1})(
+    x::NamedTuple{names2}
+) where {names1, names2}
+    exprs = []
+    for n in names2
+        if n in names1
+            # Use processed value
+            push!(exprs, :($n = b.bs.$n(x.$n)))
+        else
+            # Use existing value
+            push!(exprs, :($n = x.$n))
+        end
+    end
+    return :($(exprs...), )
+end
+
+@generated function Base.inv(b::NamedBijector{names}) where {names}
+    return :(NamedBijector(($([:($n = inv(b.bs.$n)) for n in names]...), )))
+end
+
+@generated function logabsdetjac(b::NamedBijector{names}, x::NamedTuple) where {names}
+    exprs = [:(logabsdetjac(b.bs.$n, x.$n)) for n in names]
+    return :(+($(exprs...)))
+end
+
+
+######################
+### `NamedInverse` ###
+######################
+"""
+    NamedInverse <: AbstractNamedBijector
+
+Represents the inverse of a `AbstractNamedBijector`, similarily to `Inverse` for `Bijector`.
+
+See also: [`Inverse`](@ref)
+"""
+struct NamedInverse{B<:AbstractNamedBijector} <: AbstractNamedBijector
+    orig::B
+end
+Base.inv(nb::AbstractNamedBijector) = NamedInverse(nb)
+Base.inv(ni::NamedInverse) = ni.orig
+
+logabsdetjac(ni::NamedInverse, y::NamedTuple) = -logabsdetjac(inv(ni), ni(y))
+
+##########################
+### `NamedComposition` ###
+##########################
+"""
+    NamedComposition <: AbstractNamedBijector
+
+Wraps a tuple of array of `AbstractNamedBijector` and implements their composition.
+
+This is very similar to `Composed` for `Bijector`, with the exception that we do not require
+the inputs to have the same "dimension", which in this case refers to the *symbols* for the
+`NamedTuple` that this takes as input.
+
+See also: [`Composed`](@ref)
+"""
+struct NamedComposition{Bs} <: AbstractNamedBijector
+    bs::Bs
+end
+
+# Essentially just copy-paste from impl of composition for 'standard' bijectors,
+# with minor changes here and there.
+composel(bs::AbstractNamedBijector...) = NamedComposition(bs)
+composer(bs::AbstractNamedBijector...) = NamedComposition(reverse(bs))
+∘(b1::AbstractNamedBijector, b2::AbstractNamedBijector) = composel(b2, b1)
+
+inv(ct::NamedComposition) = NamedComposition(reverse(map(inv, ct.bs)))
+
+function (cb::NamedComposition{<:AbstractArray{<:AbstractNamedBijector}})(x)
+    @assert length(cb.bs) > 0
+    res = cb.bs[1](x)
+    for b ∈ Base.Iterators.drop(cb.bs, 1)
+        res = b(res)
+    end
+
+    return res
+end
+
+(cb::NamedComposition{<:Tuple})(x) = fold(|>, cb.bs; init=x)
```

Comment:
Of course, there was a typo...
```suggestion
(cb::NamedComposition{<:Tuple})(x) = foldl(|>, cb.bs; init=x)
```